### ISSUE TITLE
T-SNE visualization of AdaCode codebook

### ISSUE THREAD

**Issue by xiaofeixia2022** [kechunl/AdaCode#4]
Hello author, this is an excellent piece of work. Could you provide the implementation code for the T-SNE  visualization results in the paper? @kechunl

**Comment by xiaofeixia2022** [comment]
![image](https://github.com/kechunl/AdaCode/assets/111624970/d4116d68-f025-4ecf-b986-efac0d4a90a2)
t-SNE result like this @kechunl

**Comment by kechunl** [comment]
Hi @xiaofeixia2022 , for this visualization, we simply called `TSNE` from sklearn.manifold, with 2 components and init method as _PCA_, and visualized the results with their codebook number as the labels.

**Comment by xiaofeixia2022** [comment]
Thank you for your response! Could you please provide the source code for drawing the distribution chart? @kechunl

**Comment by kechunl** [comment]
Please check this notebook for reference: https://github.com/kechunl/AdaCode/blob/22a6eae2745fe24e8a8aeef4d4490351e72d7047/VisToyExample.ipynb
